ga4gh/decorators.py

```diff
@@ -0,0 +1,92 @@
+"""
+General purpose decorators
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import functools
+import signal
+import time
+
+
+class TimeoutException(Exception):
+    """
+    A process has taken too long to execute
+    """
+
+
+class Repeat(object):
+    """
+    A decorator to use for repeating a tagged function.
```

Comment:
That'd be a neat trick; I wouldn't be sure how to do it.  Even if we could, that might require a lot of hackery, and might not test all of the functionality we want by virtue of hacking around stuff.  What do you have against sockets?